woocommerce/woocommerce-gutenberg-products-block
# Clean the top-level of the /build folder

We are filtering the `Inner Blocks` by checking the `parent` field of the JSON metadata. If a `parent` field is defined then it's definitely an `Inner Block`.

We chose the simple solution of generating all of our Inner Blocks' JSON files in a single folder called `inner-blocks`. This solution will clean the `/build` folder and allow us to support any newly created `Inner Blocks` by default, eliminating the need to manually fix them (e.g., Generating JSON files inside a `mini-cart`, `cart-blocks`, `checkout-blocks`, etc. folders).


Fixes #6972




#### Accessibility




#### Other Checks


### Screenshots



| Before | After |
| ------ | ----- |
|  <img width="432" alt="image" src="https://user-images.githubusercontent.com/14235870/190516228-c7ddb089-5e3b-41f4-af1a-1c534d7e5bd0.png">  |  <img width="363" alt="image" src="https://user-images.githubusercontent.com/14235870/190516587-5dff7267-9468-4eec-b142-0661f83a95d3.png"> |

### Testing

To test this PR:

1. Checkout to the `clean/6972-build-folder` branch
2. Execute the command `npm run build`.
3. Open the `/build` folder. Check that there aren't any generated Inner Blocks folders (e.g., `cart-items-block`, `checkout-totals-block`, etc.). Additionally, ensure that the `inner-blocks` folder was created.
4. Open the `/build/inner-blocks` folder. Check that the Inner Blocks folders with their `block.json` files were generated (Please see image below).

<img width="337" alt="image" src="https://user-images.githubusercontent.com/14235870/190518197-7f2e2dd0-e243-435d-9543-8c8cefff970f.png">

5. Please go through the `User Facing Testing` instructions, and ensure that the translations are still working as expected

#### Automated Tests

#### User Facing Testing



1. Change your site language (I tested using French & NL)
2. Go to the C&C Blocks (Editor mode)
3. Select each inner block and make sure the `title` & `description` are translated in the inspector. See image below:

![image](https://user-images.githubusercontent.com/14235870/181719689-f9decbdd-5bc6-4192-8612-4b2696a8d7c4.png)


### WooCommerce Visibility




### Performance Impact
